central/sensor/service/connection/worker_queue.go

```diff
@@ -55,8 +65,26 @@ func (w *workerQueue) push(msg *central.MsgFromSensor) {
 func (w *workerQueue) runWorker(ctx context.Context, idx int, stopSig *concurrency.ErrorSignal, handler func(context.Context, *central.MsgFromSensor) error) {
 	queue := w.queues[idx]
 	for msg := queue.pullBlocking(stopSig); msg != nil; msg = queue.pullBlocking(stopSig) {
-		if err := handler(ctx, msg); err != nil {
-			log.Errorf("Error handling sensor message: %v", err)
+		err := handler(ctx, msg)
+		if err != nil {
+			if pgutils.IsTransientError(err) {
+				msg.ProcessingAttempt++
+
+				if msg.ProcessingAttempt == maxHandlerAttempts {
+					log.Errorf("Error handling sensor message %T permanently: %v", msg.GetEvent().GetResource(), err)
+					continue
+				}
+				time.AfterFunc(time.Duration(msg.ProcessingAttempt)*handlerRetryInterval, func() {
```

Comment:
I don't know about the error rate on sensor messages, but since `time/concurrency.AfterFunc` spawns a go routine for each message we want to reprocess, it might be worthwhile limiting the number of go routines spawned per worker queue via a semaphore.

Reply:
My thought was that this should be rare. time.AfterFunc apparently has some performance improvements where timers are grouped in the runtime, I will see if a generic time.Timer follows the same codepath

Reply:
I'll run a scale test with a lot of failures to see how expensive this is, but I generally believe timers should be fairly optimized

Reply:
Yeah seems like the generic `timer.Timer` shares the same path as `time.AfterFunc`, sounds good to me then.